config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/ServiceLoaderTest.java

```diff
@@ -58,6 +58,125 @@ void load() {
         assertPropertiesContent(expected, service.getProperties());
     }
 
+    @Test
+    void soapServiceLoad() {
+        ServiceLoader loader = new ServiceLoader();
+        Bundle bundle = new Bundle();
+        loader.load(bundle, createCommonXml(
+                DocumentTools.INSTANCE.getDocumentBuilder().newDocument(), true
+        ));
+
+        assertFalse(bundle.getServices().isEmpty());
+        assertEquals(1, bundle.getServices().size());
+        Service service = bundle.getServices().get("service");
+        assertNotNull(service);
+        assertEquals("service", service.getName());
+        assertEquals("/soap-service", service.getUrl());
+        assertEquals("soapId", service.getId());
+        assertNotNull(service.getServiceDetailsElement());
+        assertFalse(service.getHttpMethods().isEmpty());
+        assertEquals(4, service.getHttpMethods().size());
+        assertTrue(service.getHttpMethods().containsAll(Arrays.asList("GET", "POST", "PUT", "DELETE")));
+        assertNotNull(service.getParentFolder());
+        assertEquals("folder", service.getParentFolder().getId());
+        assertEquals("wsdl file", service.getWsdl().getWsdlXml());
+        assertEquals("1.1", service.getWsdl().getSoapVersion());
+        Map<String, Object> expected = new HashMap<>();
+        expected.put("prop", "value");
+        expected.put("ENV.prop", null);
+        assertPropertiesContent(expected, service.getProperties());
+    }
+
+    private static Element createCommonXml(Document document, boolean isSoap) {
```

Comment:
It's good the methods here, I just think you could do some refactoring adding parameters to the method so you end with only one method building the xml and less duplicated stuff.

Reply:
I did not understand you.. This is common method for both tests.

Reply:
Sorry I got confused by the github view, I pulled your branch and I see you are using the same method for both tests, but there are unused methods in the class so remove them and it's fine =)